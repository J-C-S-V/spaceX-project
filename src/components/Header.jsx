import { NavLink } from 'react-router-dom';
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import log from '../assets/logo.png';

function Header() {
  const activeStyle = {
    textDecoration: 'underline',
  };

  return (
    <Navbar bg="light" expand="lg">
      <Container className="d-flex">
        <Nav.Link className="pr-2">
          <img src={log} width="100px" alt="logo" />
        </Nav.Link>
        <Navbar.Brand className="justify-self-start" href="#home">
          Space Traveler&apos;s Hub
        </Navbar.Brand>
        <Nav className="justify-content-end flex-grow-1">
          <Nav.Link href="#home">Rockets</Nav.Link>
          <NavLink
            to="/Missions"
            activeClassName="active"
            style={activeStyle}
            className="nav-link"
          >
            Missions
          </NavLink>
          <NavLink
            to="/Profile"
            activeClassName="active"
            style={activeStyle}
            className="nav-link"
          >
            My Profile
          </NavLink>
        </Nav>
      </Container>
    </Navbar>
  );
}

export default Header;
