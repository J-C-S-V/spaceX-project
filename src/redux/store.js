import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './missionsSlice';

const store = configureStore({
  reducer: {

    mission: missionReducer,

  },
});

export default store;
