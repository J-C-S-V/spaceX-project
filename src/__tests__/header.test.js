import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Header from '../components/Header';
import store from '../redux/store';

describe('renders component properly', () => {
  test('renders Header component', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <Header />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
  test('displays the logo properly', () => {
    const { getByAltText } = render(
      <BrowserRouter>
        <Provider store={store}>
          <Header />
        </Provider>
      </BrowserRouter>,
    );
    const logoElement = getByAltText('logo');
    expect(logoElement).toMatchSnapshot();
  });
});
