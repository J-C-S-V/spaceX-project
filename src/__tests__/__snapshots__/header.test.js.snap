// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component properly displays the logo properly 1`] = `
<img
  alt="logo"
  src="logo.png"
  width="100px"
/>
`;

exports[`renders component properly renders Header component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-lg navbar-light bg-light"
      >
        <div
          class="d-flex container"
        >
          <a
            class="pr-2 nav-link"
            href="#"
            role="button"
            tabindex="0"
          >
            <img
              alt="logo"
              src="logo.png"
              width="100px"
            />
          </a>
          <a
            class="justify-self-start navbar-brand"
            href="#home"
          >
            Space Traveler's Hub
          </a>
          <div
            class="justify-content-end flex-grow-1  mt-4 mr-4 p-3 navbar-nav"
          >
            <a
              aria-current="page"
              class="pe-3 active"
              href="/"
            >
              Rockets
            </a>
            <a
              class="pe-3"
              href="/Missions"
            >
              Missions
            </a>
            <a
              class=""
              href="/Profile"
            >
              My Profile
            </a>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-lg navbar-light bg-light"
    >
      <div
        class="d-flex container"
      >
        <a
          class="pr-2 nav-link"
          href="#"
          role="button"
          tabindex="0"
        >
          <img
            alt="logo"
            src="logo.png"
            width="100px"
          />
        </a>
        <a
          class="justify-self-start navbar-brand"
          href="#home"
        >
          Space Traveler's Hub
        </a>
        <div
          class="justify-content-end flex-grow-1  mt-4 mr-4 p-3 navbar-nav"
        >
          <a
            aria-current="page"
            class="pe-3 active"
            href="/"
          >
            Rockets
          </a>
          <a
            class="pe-3"
            href="/Missions"
          >
            Missions
          </a>
          <a
            class=""
            href="/Profile"
          >
            My Profile
          </a>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
